### PR TITLE
WIP: User unsubscribe 286

### PR DESCRIPTION
**Pull request by groovecoder**
This is a very ugly start but I'd like to get a review.

### CODE REVIEW

File: views/unsubscribe.hbs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{{!< default }}
+<div class="unsubscribe-message-wrapper">
+  <div class="unsubscribe-message">
+      {{#if unsubscribed }}
+          <h4>You have unsubscribed.</h4>
```
[the remark this answers is not in the document]
Comment (groovecoder) [reply views/unsubscribe.hbs]: Yeah the email and page text and designs are all just WIP right now. This PR is just to get them into place so @lesleyjanenorton can make them better.

File: views/unsubscribe.hbs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      {{ else }}
+          <h4>Are you sure you want to unsubscribe?</h4>
+          <form action="/user/unsubscribe" method="post" id="unsubscribe-form">
```
[the remark this answers is not in the document]
Comment (groovecoder) [reply views/unsubscribe.hbs]: Yeah the email and page text and designs are all just WIP right now. This PR is just to get them into place so @lesleyjanenorton can make them better.

**Comment by groovecoder** [comment]
All comments addresses except the couple that will be fixed with final email designs & HTML.

**Comment by pdehaan** [comment]
Not sure if you want both commits in this PR, or if you want to remove first one to avoid merging in the refresh breaches code as well

**Comment by groovecoder** [comment]
Removed the breaches endpoint commit.

File: db/DB.js [hunk 3]
```diff
@@ -84,17 +94,15 @@ const DB = {
...
       .update({ verified: true })
       .returning("*");
     return verifiedSubscriber;
   },
 
-  async removeSubscriber(email) {
+  async removeSubscriberByEmail(email) {
     const sha1 = getSha1(email);
-
     return await this._getSha1EntryAndDo(sha1, async aEntry => {
```
Comment (pdehaan) [inline db/DB.js]: I guess outside of the scope of this PR, but...
Why does it take 2 queries to delete a user? By proxying `this._getSha1EntryAndDo(sha1)`, we call `knex("subscribers").where("sha1", sha1)`, and then here we requery the same table by `knex("subscribers").where("id", "=", aEntry.id).del()`

Comment (groovecoder) [reply db/DB.js]: Yes, another refactor I'd like to perform is to remove the `_getSha1EntryAndDo` function completely.

**Pull request merged**